### PR TITLE
User/lobourre/update to v1

### PR DESCRIPTION
**Pull request by LPBourret**
- Update all skill samples (sample skill, sample apps in all frameworks) to non-preview release, all documentation pages (samples, APIs, intelisense)
- Rename *Windows Vision Skills* to *AI Skills for Windows* across the board and preview namespaces
- Update nuget package dependencies (C++WinRT, NetCore.UniversalWindows, Widows Community Toolkit, VCRT Forwarder, etc.) and build scripts
- Update sample skill (FaceSentimentAnalyzer) to detect and classify multiple faces in a single evaluation

Release note draft for v1.0:
- Rename Windows Vision Skills to AI Skills for Windows to reflect the broader scope of the framework
- Add support for DXCore execution devices (i.e. Intel VPU)
- Remove the preview moniker for the interface and the set of released AI Skills from Microsoft. Preview nuget package are deprecated and now point to their non-preview counterparts
- Added feature sourcing (ISkillFeature.SourceFromOtherFeature()), allowing a feature to be sourced from another when the same value is usually used, bypassing the need to update twice before every evaluation and ensuring optimal processing.
- Add support for SkillFeatureCustomTensor type to encapsulate a tensor of arbitrary primitives
- SkillFeatureImageDescriptor: Remove MaxDimension property
- Allow to pass a shape value below -1 in SkillFeatureTensorDescriptor to specify a required factor (i.e. [-1, 2] means a shape divisible by 2 is required, if the size of the value provided does not match, it is clipped by the closest such value). Same thing for width and height value defined by a SkillFeatureImageDescriptor, where image is center-cropped to match the factor.
- Binary size reduction across the board
- Updated documentation and samples
- Misc improvements across the board (bug fixes, use latest C++WinRT, etc.)

### CODE REVIEW

**Comment by NaveenThum** [comment]
@chrilaMSFT is added to the review. #Closed

**Comment by NaveenThum** [comment]
@saredd is added to the review. #Closed

**Comment by NaveenThum** [comment]
@Sudhaushu is added to the review. #Closed

File: samples/SentimentAnalyzerCustomSkill/cpp/Skill/FaceSentimentAnalyzer/Contoso.FaceSentimentAnalyzer.vcxproj [hunk 1]
```diff
@@ -147,6 +156,15 @@
       <Message Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">Generating manifest file for the WindowsRuntimeComponent to be merged into desktop app manifest for automatic registration of components</Message>
       <Message Condition="'$(Configuration)|$(Platform)'=='Release|x64'">Generating manifest file for the WindowsRuntimeComponent to be merged into desktop app manifest for automatic registration of components</Message>
     </PostBuildEvent>
+    <Link>
+      <AdditionalDependencies Condition="'$(Configuration)|$(Platform)'=='Release|ARM'">kernel32.lib;ole32.lib;OleAut32.lib;WindowsApp.lib;%(AdditionalDependencies)</AdditionalDependencies>
```
Comment (NaveenThum) [inline samples/SentimentAnalyzerCustomSkill/cpp/Skill/FaceSentimentAnalyzer/Contoso.FaceSentimentAnalyzer.vcxproj:160]: >ARM [](start = 82, length = 3)

Is ARM64 broken for this skill even on CPU?

**Comment by NaveenThum** [comment]
Which skills are we going to leave with Preview in them?

**Comment by NaveenThum** [comment]
Sentiment analysis - how is the updated skill behaving with sample customer images?

**Comment by NaveenThum** [comment]
In the change description, "i.e. [-1, 2] means" - should this be -2 means?

File: samples/Common/cs/FrameReaderFrameSource.cs [hunk 2]
```diff
@@ -168,12 +168,12 @@ private async Task InitializeMediaFrameSourceAsync()
             int preferredFrameWidth = 1920;
             if (m_desiredImageDescriptor != null && m_desiredImageDescriptor.Width != -1)
             {
-                preferredFrameWidth = m_desiredImageDescriptor.Width;
+                preferredFrameWidth = Math.Abs(m_desiredImageDescriptor.Width);
```
Comment (NaveenThum) [inline samples/Common/cs/FrameReaderFrameSource.cs:171]: >Math.Abs [](start = 38, length = 8)

What is the meaning for a width = -640?

File: README.md [hunk 3]
```diff
@@ -2,39 +2,48 @@
...
+Implementing and integrating efficient AI and Computer Vision (CV) solutions is a hard task for developers. The industry is moving at a fast pace and the amount of custom-tailored solutions coming out makes it almost impossible for app developers to keep up easily. 
+
+This proposed framework is meant to standardize the way AI and CV is put to use within a Windows application (i.e.: UWP, Desktop Win32, .Net Core 3.0+) running on the edge. It aims to abstract away the complexity of AI techniques by simply defining the concept of *skills* which are modular pieces of code that process input(s) and produce output(s). The implementation that contains the complex details is encapsulated by extended WinRT API that inherits the base class present in the Microsoft.AI.Skills.SkillInterface namespace, which leverages built-in Windows primitives which in-turn eases interop with built-in hardware acceleration leveraged by frameworks such as *[Windows ML](https://docs.microsoft.com/en-us/windows/ai/windows-ml/)*.
```
Comment (NaveenThum) [inline README.md:7]: >proposed [](start = 5, length = 8)

proposed?

File: README.md [hunk 4]
```diff
@@ -2,39 +2,48 @@
...
+For how to use the AI Skills interface to author a new AI Skill of your own exposing your AI solution to other Windows app developers, and creating an actual app to consume this crafted skill, see this sample:
 
-For how to use the framework to author a Skill to be consumed, and creating an app to consume the skill, see the [SentimentAnalyzerCustomSkill](samples/SentimentAnalyzerCustomSkill) Sample. 
+### **[Sentiment Analyzer Sample Skill](samples/SentimentAnalyzerSampleSkill)**
+
+| ![FaceSentimentAnalyzer sample skill logo](./doc/FaceSentimentAnalyzerLogo.png) | Walking through the steps to create, in C++ or in C#, a sample AI Skill that leverages WinML and MediaFoundation as well as how to consume it in a Win32 Dektop, .Net Core 3.0+ or UWP app|
+| -- | -- |
 
-## Samples for AI skills [published by Microsoft on nuget.org](https://www.nuget.org/profiles/VisionSkills)
+## Code samples for using AI skills [published by Microsoft on nuget.org](https://www.nuget.org/profiles/VisionSkills)
```
Comment (NaveenThum) [inline README.md:22]: >VisionSkills [](start = 105, length = 12)

Should we reserve and move to AISkills?

**Comment by NaveenThum** [comment]
Very nice and clean update!

**Review by NaveenThum — DISMISSED** [review]
:shipit:

File: README.md (hunk 4 above)
Comment (LPBourret) [reply README.md:22]: it is just the name of our nuget publishing org, yes we can submit for a new one, not sure if we can rename an nuget publishing organization

File: samples/Common/cs/FrameReaderFrameSource.cs (hunk 2 above)
Comment (LPBourret) [reply samples/Common/cs/FrameReaderFrameSource.cs:171]: a width divisible by 640

Comment (LPBourret) [reply samples/Common/cs/FrameReaderFrameSource.cs:171]: that is in the context of a SkillFeatureImageDescriptor.Width, in the context of this app consuming such object, we just look for the closest available resolution to the abs() of this value, I could change that logic to accommodate negative values properly

File: README.md (hunk 4 above)
Comment (LPBourret) [reply README.md:22]: > Just curious: Sentiment analysis - how is the updated skill behaving with sample customer images?

we can sync offline

**Comment by LPBourret** [comment]
> Aren't we going to leave some skills as 'preview'?

yes the private ones never published yet, we can sync offline

**Comment by LPBourret** [comment]
> In the change description, "i.e. [-1, 2] means" - should this be -2 means?

for images where is a single dimension mentioned, we take single negative value to express a factor. (i.e. width == -2 means I need an image with a width divisable by 2 or I else will clip a column of pixel out of your odd-shaped image upon binding). For Tensors, since they get flatten out, we accept a single negative value followed by other dimensions to express a factor. i.e. we don't accept [-2, -6], I made the choice to defer to the existing free dimension value (-1) to express this constraint. I could probably have looked for if the shape specified is a single value and if it is negative for redundancy.

File: samples/SentimentAnalyzerCustomSkill/cs/Skill/FaceSentimentAnalyzer/FaceSentimentAnalyzerBinding.cs [hunk 5]
```diff
@@ -68,68 +68,56 @@ public bool IsFaceFound
...
                 }
-                
-                return predominantSentiment;
+                return predominantSentiments;
             }
         }
 
         /// <summary>
-        /// Returns the face rectangle
+        /// Returns the bounding box around each detected faces
```
Comment (saredd) [inline samples/SentimentAnalyzerCustomSkill/cs/Skill/FaceSentimentAnalyzer/FaceSentimentAnalyzerBinding.cs:112]: each detected face

Comment (saredd) [reply samples/SentimentAnalyzerCustomSkill/cs/Skill/FaceSentimentAnalyzer/FaceSentimentAnalyzerBinding.cs:112]: There was somewhere else with the same text that should also be changed but having trouble finding it now :(

File: doc/Microsoft.AI.Skills.SkillInterface.md [hunk 6]
```diff
@@ -1664,6 +1841,7 @@ The SkillFeatureImageDescriptor instantiated.
 ##### CreateValueAsync() <a name="SkillFeatureImageDescriptor.CreateValueAsync"></a>
 
 Create a [ISkillFeatureValue](#ISkillFeatureValue) of [SkillFeatureKind](#SkillFeatureKind) *Image* according to the format specified.
+Acceptable value types are [VideoFrame][VideoFrame], ISkillFeatureImageValue.
```
[the remark this answers is not in the document]
Comment (LPBourret) [reply doc/Microsoft.AI.Skills.SkillInterface.md:1844]: no the use of # refers to an anchor in the page, while in this case it refers to the link associated with the [VideoFrame] markup stored at the bottom of this page (line 1955)

---
In reply to: [412476237](https://github.com/microsoft/AISkillsForWindows/pull/68#discussion_r412476237) [](ancestors = 412476237)

File: doc/Microsoft.AI.Skills.SkillInterface.md [hunk 7]
```diff
@@ -1029,7 +1029,7 @@ All SkillExecutionDeviceDirectX available on the system.
 ### SkillExecutionDeviceDXCore <a name="SkillExecutionDeviceDXCore"></a>
 ``implements`` [ISkillExecutionDeviceDX](#ISkillExecutionDeviceDX)
 
-Provides a DirectX execution device based on [DXCore][DXCore] API set and its information useful to infer if a skill could be run with it appropriately in the consumer's context. 
+Provides a [DXCore][DXCore] execution device and its information useful useful to infer if a skill could be run with it appropriately in the consumer's context.
```
Comment (saredd) [inline doc/Microsoft.AI.Skills.SkillInterface.md:1032]: duplicate useful

**Review by NaveenThum — APPROVED** [review]
:shipit:

**Pull request merged**